lib/travis/services/find_build.rb

```diff
@@ -31,8 +32,16 @@ def all_resources
           end
         end
 
-        def result
-          @result ||= scope(:build).find_by_id(params[:id])
+        def result(options = {})
+          @result ||= load_result(options)
+        end
+
+        def load_result(options = {})
+          columns = scope(:build).column_names
+          columns -= %w(config) if options[:exclude_config]
+          scope(:build).select(columns).find_by_id(params[:id]).tap do |res|
+            res.config = {} if options[:exclude_config]
```

Comment:
By default it's not merely `nil`, but an unresolved relation.  Other parts of the code (somewhere!?!) want to treat it as a hash to default in `language` and `os`, but without making it a hash first we end up with ActiveRecord explosions.